src/core/mount-setup.c

```diff
@@ -397,9 +397,19 @@ int mount_setup(bool loaded_policy) {
          * nspawn and the container tools work out of the box. If
          * specific setups need other settings they can reset the
          * propagation mode to private if needed. */
-        if (detect_container() <= 0)
-                if (mount(NULL, "/", NULL, MS_REC|MS_SHARED, NULL) < 0)
-                        log_warning_errno(errno, "Failed to set up the root directory for shared mount propagation: %m");
+        if (mount(NULL, "/", NULL, MS_REC|MS_SHARED, NULL) < 0)
+                log_warning_errno(errno, "Failed to set up the root directory for shared mount propagation: %m");
+
```

Comment:
Yes, I just set MS_SHARED unconditionally (both for systemd on the host or systemd in a container), independently of `$container`. Do you know of any issue with that?

Note that containers will still have their "/" as slave to the host since it is not possible on Linux to "unslave" a mount to rejoin the peer group where it was previously. In that scenario, this patch will just make the mounts slave & shared instead of just slave.